modules/apps/account/account-service/src/main/java/com/liferay/account/service/impl/AccountRoleLocalServiceImpl.java

```diff
@@ -274,35 +276,49 @@ public BaseModelSearchResult<AccountRole> searchAccountRoles(
 		LinkedHashMap<String, Object> params, int start, int end,
 		OrderByComparator<?> orderByComparator) {
 
-		SearchResponse searchResponse = _searcher.search(
-			_getSearchRequest(
-				companyId, accountEntryIds, keywords, params, start, end,
-				orderByComparator));
+		long accountEntryId = 0;
 
-		SearchHits searchHits = searchResponse.getSearchHits();
+		if (accountEntryIds.length > 0) {
+			accountEntryId = accountEntryIds[0];
+		}
 
-		return new BaseModelSearchResult<AccountRole>(
-			TransformUtil.transform(
-				searchHits.getSearchHits(),
-				searchHit -> {
-					Document document = searchHit.getDocument();
+		try (SafeCloseable safeCloseable =
+				AccountRolePermissionThreadLocal.setWithSafeCloseable(
+					accountEntryId)) {
 
```

Comment:
why do you need this?

Reply:
To make available the accountEntryId in the PermissionChecker